packages/builders/src/interactions/commands/chatInput/Assertions.ts

```diff
@@ -0,0 +1,139 @@
+
+export const sharedNameAndDescriptionPredicate = z.object({
+	name: namePredicate,
+	name_localizations: localeMapPredicate.optional(),
+	description: descriptionPredicate,
+	description_localizations: localeMapPredicate.optional(),
+});
+
+export const numericMixinNumberOptionPredicate = z.object({
+	max_value: z.number().optional(),
+	min_value: z.number().optional(),
+});
+
+export const numericMixinIntegerOptionPredicate = z.object({
+	max_value: z.number().int().optional(),
+	min_value: z.number().int().optional(),
+});
+
+export const channelMixinOptionPredicate = z.object({
+	channel_types: z
+		.union(
+			applicationCommandOptionAllowedChannelTypes.map((type) => z.literal(type)) as unknown as [
+				ZodTypeAny,
+				ZodTypeAny,
+				...ZodTypeAny[],
+			],
+		)
+		.array()
+		.optional(),
+});
+
+export const autocompleteMixinOptionPredicate = z.object({
+	autocomplete: z.literal(true),
+	choices: z.union([z.never(), z.never().array(), z.undefined()]),
+});
+
+export const choiceStringPredicate = z.string().min(1).max(100);
+export const choiceNumberPredicate = z.number().min(Number.NEGATIVE_INFINITY).max(Number.POSITIVE_INFINITY);
+export const choicePredicate = z.object({
+	name: choiceStringPredicate,
+	name_localizations: localeMapPredicate.optional(),
+	value: z.union([choiceStringPredicate, choiceNumberPredicate]),
+});
+
+export const choicesOptionMixinPredicate = z.object({
+	autocomplete: z.literal(false).optional(),
+	choices: choicePredicate.array().max(25).optional(),
+});
+
+export const basicOptionTypes = [
+	ApplicationCommandOptionType.Attachment,
+	ApplicationCommandOptionType.Boolean,
+	ApplicationCommandOptionType.Channel,
+	ApplicationCommandOptionType.Integer,
+	ApplicationCommandOptionType.Mentionable,
+	ApplicationCommandOptionType.Number,
+	ApplicationCommandOptionType.Role,
+	ApplicationCommandOptionType.String,
+	ApplicationCommandOptionType.User,
+] as const;
+
+export const basicOptionTypesPredicate = z.union(
+	basicOptionTypes.map((type) => z.literal(type)) as unknown as [ZodTypeAny, ZodTypeAny, ...ZodTypeAny[]],
+);
+
+export const basicOptionPredicate = sharedNameAndDescriptionPredicate.extend({
+	required: z.boolean().optional(),
+	type: basicOptionTypesPredicate,
+});
+
+export const autocompleteOrChoicesMixinOptionPredicate = z.discriminatedUnion('autocomplete', [
+	autocompleteMixinOptionPredicate,
+	choicesOptionMixinPredicate,
+]);
+
+export const channelOptionPredicate = basicOptionPredicate.merge(channelMixinOptionPredicate);
+
+export const integerOptionPredicate = basicOptionPredicate
+	.merge(numericMixinIntegerOptionPredicate)
+	.and(autocompleteOrChoicesMixinOptionPredicate);
+
+export const numberOptionPredicate = basicOptionPredicate
+	.merge(numericMixinNumberOptionPredicate)
+	.and(autocompleteOrChoicesMixinOptionPredicate);
+
+export const stringOptionPredicate = basicOptionPredicate
+	.extend({
+		max_length: z.number().optional(),
+		min_length: z.number().optional(),
```

Comment:
https://github.com/discord/discord-api-spec/blob/348aeb6011459285685c9a4db7989a6fd4cbf508/specs/openapi.json#L11461-L11476

```suggestion
		max_length: z.number().min(0).max(6_000).optional(),
		min_length: z.number().min(1).max(6_000).optional(),
```